tests/test_search_pattern.py

```diff
@@ -0,0 +1,187 @@
+"""
+Tests the SearchPattern class
+"""
+
+import pytest
+
+from modules import decision_command
+from modules import odometry_and_time
+from modules import drone_odometry_local
+from modules.decision import search_pattern
+
+DISTANCE_SQUARED_THRESHOLD = 2.0
+SMALL_ADJUSTMENT = 1
+
+
+@pytest.fixture()
+def search_pattern_width_greater_depth():
+    """
+    Initializes a SearchPattern instance.
+    """
+    search_pattern_instance = search_pattern.SearchPattern(
+        camera_fov_forwards=30.0,
+        camera_fov_sideways=60.0,
+        search_height=100.0,
+        search_overlap=0.2,
+        current_position_x=100.0,
+        current_position_y=50.0,
+        distance_squared_threshold=DISTANCE_SQUARED_THRESHOLD,
+        small_adjustment=SMALL_ADJUSTMENT,
+    )
+    yield search_pattern_instance
+
+
+@pytest.fixture()
+def search_pattern_depth_greater_width():
+    """
+    Initializes a SearchPattern instance.
+    """
+    search_pattern_instance = search_pattern.SearchPattern(
+        camera_fov_forwards=60.0,
+        camera_fov_sideways=30.0,
+        search_height=50.0,
+        search_overlap=0.2,
+        current_position_x=75.0,
+        current_position_y=150.0,
+        distance_squared_threshold=DISTANCE_SQUARED_THRESHOLD,
+        small_adjustment=SMALL_ADJUSTMENT,
+    )
+    yield search_pattern_instance
+
+
+@pytest.fixture()
+def drone_odometry():
+    def create_drone_odometry(pos_x, pos_y, depth):
+        """
+        Creates an OdometryAndTime instance representing the drone's current position.
+        """
+        position = drone_odometry_local.DronePositionLocal.create(pos_y, pos_x, -depth)[
+            1
+        ]
+        orientation = drone_odometry_local.DroneOrientationLocal.create_new(
+            0.0, 0.0, 0.0
+        )[1]
+        odometry_data = drone_odometry_local.DroneOdometryLocal.create(
+            position, orientation
+        )[1]
+
+        result, odometry = odometry_and_time.OdometryAndTime.create(odometry_data)
+        return odometry
+
+    return create_drone_odometry
+
+
+def assert_move_command(move, newPos, target_x, target_y, target_z):
+    assert move[0] == newPos
+    assert (
+        move[1].get_command_type()
+        == decision_command.DecisionCommand.CommandType.MOVE_TO_ABSOLUTE_POSITION
+    )
+    pos = move[1].get_command_position()
+    assert pytest.approx(target_x, 0.1) == pos[0]
+    assert pytest.approx(target_y, 0.1) == pos[1]
+    assert pytest.approx(target_z, 0.1) == pos[2]
```

Comment:
set this up with the testing convention
1. inputs and expected - expected_move = newPos, etc.
2. actual values - actual_pos = move[0]
3. assert actual == expected